spec/configSpec.js

```diff
@@ -118,5 +118,185 @@ describe('config', function() {
         validateConfig(config);
       }).toThrowError("Configuration's helpers property is not an array");
     });
+
+    describe('config.importMap', () => {
+      it('throws if the importMap property is not an object', function() {
+        expect(function() {
+          const config = validConfig();
+          config['importMap'] = 'just a string';
+          validateConfig(config);
+        }).toThrowError(/Configuration.+importMap.+is not an object/);
+      });
+
+      it('throws if the importMap property is set but has no imports or scopes', function() {
+        expect(function() {
+          const config = validConfig();
+          config['importMap'] = {};
+          validateConfig(config);
+        }).toThrowError(
+          /Configuration.+importMap.+contains no imports or scopes/
```

Comment:
Again, your opinion rules here. Personally, I've become less of a fan of matching against string literals over the years and I thought it was close enough visually and effectively to the existing ones. But I'll go ahead and change it over if you feel comfortable with all of the error messages (those messages could have been something you wanted to change in code review for example, but I couldn't know your opinion on those beforehand!).

Reply:
I've changed this to string literals in [this commit](https://github.com/jasmine/jasmine-browser-runner/pull/30/commits/27b1e96c31ead16a991dc93a7949b4fa2d25f79e).

Additionally, this only took a couple minutes...but it was a painful couple of minutes. This was precisely the kind of stylistic opinion I would have appreciated you giving feedback on earlier to avoid having to do this. But on the other hand, I've skimmed over some of the neat stuff you've been doing in the pivotal tracker backlog and I can definitely see how you might not have had the time to notice this and form this opinion yourself before.

And to be a bit cheeky (though inflection is lost in text-only comms and we don't really know each other), no I don't think it would be simpler or clearer, though possibly more effective on a large scale but not at this scale. Personally I use UUIDs (or GUIDs depending on your religion!) because I think the overhead is worth the ease with which you can just search in a codebase for a unique sequence of characters. Otherwise, you have to parse the error message and hope someone hasn't copy/pasted too large of a portion of the error message, let alone those errors that are exactly the same strings! That said, of course I don't see you retrofitting a hugely popular and already well-established codebase to use these and as you can see from the commit, I defer to your sovereign rule here! 👑

Reply:
> Additionally, this only took a couple minutes...but it was a painful couple of minutes. This was precisely the kind of stylistic opinion I would have appreciated you giving feedback on earlier to avoid having to do this. 

I provided that feedback as soon as I saw the code in question, which I *think* is also the first time I had access to it. It's possible that there was a spec like this in one of the commits you linked to prior to opening the PR and I missed it. But in any case, I don't generally give WIP that kind of thorough, detail-oriented review. For two reasons: because I usually don't have enough time to give PRs multiple passes at that level of detail, and because a lot of developers don't appreciate getting that type of feedback when they're still trying to get things working.

What I *did* do is write all of the existing config validation specs in a consistent style. In a way, that *is* an expression of stylistic opinion. You could have avoided the rework by matching the existing code rather than solving the same problem in a different way.

I appreciate that you've ultimately been willing to conform to the local style. But there have been several cases where you've done that only after departing from it and advocating for your way of doing things. That can be ok (although maintainer reactions to it can vary greatly) but it's not a good strategy if you're looking to get a PR merged with a minimum of change requests.

That's not to say that you can't write the code the way you want to. You don't owe anyone a PR at all, let alone one that's written in a particular style. But keep in mind that "I want to do things my way" and "I want to get this merged with a minimum of fuss and rework" tend to be mutually exclusive goals that call for different mindsets.